test/components/birth-date-picker.spec.tsx

```diff
@@ -0,0 +1,141 @@
+import * as React from "react";
+import { ClientRenderer, expect, sinon, waitFor, change } from "test-drive-react";
+import { BirthDatePicker, dateFromYearMonthDay } from "../../src";
+import { BirthDatePickerDemo } from "../../demo/components/birth-date-picker-demo";
+
+describe("<BirthDatePicker />", () => {
+    const clientRenderer = new ClientRenderer();
+    afterEach(() => clientRenderer.cleanup());
+
+    it("Displays the date and allows changing it", async function() {
+        const { select, waitForDom } = clientRenderer.render(<BirthDatePickerDemo />);
+
+        await waitForDom(() => {
+            expect(select("BIRTH_DATE_PICKER")).to.be.present();
+        });
+
+        change(select("BIRTH_DATE_PICKER_YEAR"), "2002");
+        change(select("BIRTH_DATE_PICKER_MONTH"), "10");
+        change(select("BIRTH_DATE_PICKER_DAY"), "12");
+
+        await waitFor(() => {
+            expect(select("BIRTH_DATE_PICKER_DEMO_RESULT")).to.contain.text("Selected date: 2002-10-12");
+        });
+
+        change(select("BIRTH_DATE_PICKER_DAY"), "");
+        await waitFor(() => {
+            expect(select("BIRTH_DATE_PICKER_DEMO_RESULT")).to.contain.text("Selected date: 2002-10-12");
+        });
+    });
+
+    it("Renders with default props", function () {
+        const { select, waitForDom } = clientRenderer.render(<BirthDatePicker />);
+
+        return waitForDom(() => {
+            expect(select("BIRTH_DATE_PICKER")).to.be.present();
+            expect(select("BIRTH_DATE_PICKER_YEAR")).to.have.value("");
+            expect(select("BIRTH_DATE_PICKER_MONTH")).to.have.value("");
+            expect(select("BIRTH_DATE_PICKER_DAY")).to.have.value("");
+        });
+    });
+
+    it("Displays the provided date", function() {
+        const { select, waitForDom } = clientRenderer.render(
+            <BirthDatePicker value={new Date("1986-04-26Z")} />
+        );
+
+        return waitForDom(() => {
+            expect(select("BIRTH_DATE_PICKER_YEAR")).to.have.value("1986");
+            expect(select("BIRTH_DATE_PICKER_MONTH")).to.have.value("4");
+            expect(select("BIRTH_DATE_PICKER_DAY")).to.have.value("26");
+        });
+    });
+
+    it("Does not emit onChange for initial value", function() {
```

Comment:
This test is a bit problematic. Testing that something doesn't happen is usually a problem, what if the event is emitted after the timeout?